internal/expr/parser.go

```diff
@@ -473,35 +453,68 @@ func (p *Parser) parseColumn() (columnAccessor, bool, error) {
 	return columnAccessor{}, false, nil
 }
 
-func (p *Parser) parseTargetType() (valueAccessor, bool, error) {
+func (p *Parser) parseTargetType() (memberAccessor, bool, error) {
+	startLine := p.lineNum
+	startCol := p.colNum()
+
 	if p.skipByte('&') {
+		// Using a slice as an output is an error, we add the case here to
+		// improve the error message.
+		if st, ok, err := p.parseSliceAccessor(); ok {
+			return memberAccessor{}, false, errorAt(fmt.Errorf("cannot use slice syntax %q in output expression", st), startLine, startCol, p.input)
+		} else if err != nil {
+			return memberAccessor{}, false, errorAt(fmt.Errorf("cannot use slice syntax in output expression"), startLine, startCol, p.input)
+		}
 		return p.parseTypeName()
 	}
 
-	return valueAccessor{}, false, nil
+	return memberAccessor{}, false, nil
+}
+
+// parseSliceAccessor parses a slice range composed of the form "[:]".
+func (p *Parser) parseSliceAccessor() (va sliceRangeAccessor, ok bool, err error) {
+	cp := p.save()
+
+	id, ok := p.parseIdentifier()
+	if !ok {
+		return sliceRangeAccessor{}, false, nil
+	}
+	if !p.skipByte('[') {
+		cp.restore()
+		return sliceRangeAccessor{}, false, nil
+	}
+	p.skipBlanks()
+	if !p.skipByte(':') {
+		return sliceRangeAccessor{}, false, errorAt(fmt.Errorf("invalid slice: expected ':'"), p.lineNum, p.colNum(), p.input)
+	}
+	p.skipBlanks()
+	if !p.skipByte(']') {
+		return sliceRangeAccessor{}, false, errorAt(fmt.Errorf("invalid slice: expected ']'"), p.lineNum, p.colNum(), p.input)
+	}
+	return sliceRangeAccessor{typeName: id}, true, nil
 }
 
 // parseTypeName parses a Go type name qualified by a tag name (or asterisk)
 // of the form "TypeName.col_name".
-func (p *Parser) parseTypeName() (valueAccessor, bool, error) {
```

Comment:
Maybe we could change this function name to something like `parseTypeMember`, or `parseOutputType` or something.

Reply:
What about `parseTypeMemberAccessor`?

Reply:
Yea maybe, though what do you think of `parseTypeWithMember` or `parseTypeAndMember`? Since they say what it is a bit more and you can see the `memberAccessor` from the type

Reply:
I like `parseTypeAndMember`, much better than what I suggested. Changed now!